### PR TITLE
Remove CanonicalIdentificationFromImageObjectToCoimage

### PR DESCRIPTION
**Pull request by zickgraf**
See https://github.com/homalg-project/CAP_project/issues/1460

### CODE REVIEW

**Comment by mohamed-barakat** [comment]
Please do not merge until https://github.com/homalg-project/CategoricalTowers/pull/368 is merged.

**Comment by zickgraf** [comment]
> Please do not merge until [homalg-project/CategoricalTowers#368](https://github.com/homalg-project/CategoricalTowers/pull/368) is merged.

That PR is merged. (How) shall we proceed here?

**Comment by mohamed-barakat** [comment]
Please merge and we should investigate the consequences.

File: CAP/gap/DerivedMethods.gi [hunk 1]
```diff
@@ -453,22 +453,6 @@ AddDerivationToCAP( ImageEmbedding,
   
   end : CategoryFilter := IsAbelianCategory ); ##FIXME: PreAbelian?
 
-##
```
Comment (mohamed-barakat) [inline CAP/gap/DerivedMethods.gi:456]: I suggest resurrecting all these derivations with `CanonicalIdentificationFromImageObjectToCoimage` replaced by `InverseMorphismFromCoimageToImageWithGivenObjects`.

@zickgraf: Would you again run in the same problems described in https://github.com/homalg-project/CAP_project/issues/1460?

Comment (mohamed-barakat) [reply CAP/gap/DerivedMethods.gi:456]: A more refined suggestion is part of this PR https://github.com/homalg-project/CategoricalTowers/pull/377.

Comment (zickgraf) [reply CAP/gap/DerivedMethods.gi:456]: > I suggest resurrecting all these derivations with `CanonicalIdentificationFromImageObjectToCoimage` replaced by `InverseMorphismFromCoimageToImageWithGivenObjects`.
> 
> @zickgraf: Would you again run into the same problems described in #1460?

I had three problems in https://github.com/homalg-project/CAP_project/issues/1460:

1. The role of `CanonicalIdentificationFromCoimageToImageObject` compared to `MorphismFromCoimageToImage`.
2. The name `CanonicalIdentificationFromCoimageToImageObject` vs. `CanonicalIdentificationFromCoimageObjectToImageObject`.
3. Asymmetry of the derivations using `CanonicalIdentificationFromCoimageToImageObject` with regard to dualization.

The first two points are non-issues for `MorphismFromCoimageToImage`. The third would have to be investigated for `MorphismFromCoimageToImage`. `MorphismFromCoimageToImage` is currently not a CAP operation, as you noticed somewhere else, and the WithGiven version is self-dual, while `CanonicalIdentificationFromCoimageToImageObject` had a proper dual.

**Pull request merged**